datasetinsights/io/downloader/gcs_downloader.py

```diff
@@ -0,0 +1,30 @@
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.gcs import download_folder_from_gcs
+
+
+class GCSDownloader(DatasetDownloader, protocol="gs://"):
+    """ This class is used to download data from GCS
+    """
+
+    def __init__(self, **kwargs):
+        """
+
+        Args:
+            access_token: Access token to be used to authenticate to
+            unity simulation for downloading the dataset
+        """
+
+    def download(self, source_uri, output, **kwargs):
+        """
+
+        Args:
+            source_uri: This is the downloader-uri that indicates where on
+                GCS the dataset should be downloaded from.
+                The expected source-uri follows these patterns
+                usim: gs://bucket/folder
+
+            output: This is the path to the directory
+                where the download will store the dataset.
+
+        """
+        download_folder_from_gcs(source_uri, output)
```

Comment:
Could also be done with gsutil as an alternative that we should consider. Then we dont need to worry about checksums and optimizing download speed

import subprocess

`subprocess.check_call(['gsutil', '-m', 'cp','-r', source_uri, output_directory])`

Reply:
I thought of this one but creates a additional directory: Ex:
gsutil cp -r gs://my-bucket/data dir
- download from gs://my-bucket/data to be  dir/data/a/b/c. 
- Local directory start from data, which we do not want. It should be something like that dir/a/b/c
- We need to move files one layer up

Reply:
We should use GCP python [client](https://googleapis.dev/python/storage/latest/client.html#module-google.cloud.storage.client) here. Using subprocess call is usually a hack when python library is not available.

Reply:
From the comment i don't see why its a hack can you elaborate on why this would be the case?

Reply:
If the extra dir is inconvenient, you could use gsutil 'rsync' instead i think